src/PowerShellEditorServices.Host/EditorServicesHost.cs

```diff
@@ -493,6 +495,16 @@ private IServerListener CreateServiceListener(MessageProtocolType protocol, Edit
             }
         }
 
+        private static string GetOSArchitecture()
+        {
+            if (IntPtr.Size == 8)

```

Comment:
This will tell you the architecture of the current process, so this could be a x86 process running on a x64 OS.

Does `Environment.Is64BitOperatingSystem` work for us?